### PR TITLE
HOLO-678: Deployment patches

### PR DESCRIPTION
**Pull request by ACC01ADE**
## Describe Changes

Making updates to the deployment process. Primary reason is for more robust support of Super Cold Storage Signer.

- Updated deploy scripts to support the setting of gasPrice and gasLimit variables
- Created a Gas Monitoring Service for dynamic gas pricing
- Provided a way to set maximum gas fees for deployment
- Set gas pricing logic to pause deployments and wait for gas price to lower if it goes above set limits
- Updated NPM package version for networks, environment, and super-cold-storage-signer
- Including soft mainnet deployment test files
- Adding recoverJob function to Operator
- Adding tests for recoverJob function.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: deploy/11_layer_zero.ts [hunk 1]
```diff
@@ -32,11 +32,17 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
   if (currentNetworkType == NetworkType.local && lzEndpoint == zeroAddress) {
     lzEndpoint = (await hre.getNamedAccounts()).lzEndpoint;
     const mockLZEndpoint = await hre.deployments.deploy('MockLZEndpoint', {
-      from: lzEndpoint,
+      ...(await txParams({
+        hre,
+        from: lzEndpoint,
+        to: '0x0000000000000000000000000000000000000000',
```
Comment (alexanderattar) [inline deploy/11_layer_zero.ts:38]: is this is just caching current gas parameters?

Comment (ACC01ADE) [reply deploy/11_layer_zero.ts:38]: This function puts together all the data needed. Nonce, gasPrice, and gasLimit.

File: deploy/21_hlg.ts [hunk 2]
```diff
@@ -140,25 +150,48 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
...
   }
 
+  hre.deployments.log('checking HolographOperator HLG balance');
   if (currentNetworkType == NetworkType.testnet || currentNetworkType == NetworkType.localhost) {
     const hlgContract = (await hre.ethers.getContract('HolographERC20', deployer)).attach(hlgTokenAddress);
     if ((await hlgContract.balanceOf(operatorAddress)).isZero()) {
+      hre.deployments.log('HolographOperator has no HLG');
       const transferTx = await hlgContract.transfer(operatorAddress, BigNumber.from('1000000000000000000000000'), {
-        nonce: await hre.ethers.provider.getTransactionCount(deployer.address),
+        ...(await txParams({
```
Comment (alexanderattar) [inline deploy/21_hlg.ts:185]: why 2 million eth?

Comment (ACC01ADE) [reply deploy/21_hlg.ts:185]: That's just gasLimit being multiplied by 2. Final result is like 140k gas units, real used amount is 78k-ish gas units.
The other stuff you're seeing is the actual erc20 token transfer transaction being encoded for extracting gasLimit with.

Comment (ACC01ADE) [reply deploy/21_hlg.ts:185]: As for the 1mil erc20 tx, that's for our testnets so we have tokens to use for operator rewards.

File: scripts/utils/helpers.ts [hunk 3]
```diff
@@ -344,6 +348,122 @@ const genesisDeriveFutureAddress = async function (
...
+  hre: LeanHardhatRuntimeEnvironment,
+  from: string | SignerWithAddress | SuperColdStorageSigner,
+  to: string,
+  data: Promise<UnsignedTransaction> | BytesLike = '',
+  value: BigNumberish = 0
+): Promise<BigNumber> {
+  if (typeof from !== 'string') {
+    from = (from as SignerWithAddress).address;
+  }
+  if (!isBytesLike(data)) {
```
Comment (alexanderattar) [inline scripts/utils/helpers.ts]: should this check for false for clarity?

Comment (ACC01ADE) [reply scripts/utils/helpers.ts]: Good idea

Comment (ACC01ADE) [reply scripts/utils/helpers.ts]: Fixed

**Comment by ACC01ADE** [comment]
Need to merge this into `experimental` branch.

**Review by alexanderattar — APPROVED** [review]

**Pull request merged**